test/focus-trap-react.test.js

```diff
@@ -317,6 +323,219 @@ describe('FocusTrap', () => {
+    });
+
+    it('Will call onPostActivate() even if checkCanFocusTrap() is undefined', async () => {
+      const onActivate = jest.fn();
+      const onPostActivate = jest.fn();
+
+      render(
+        <FocusTrapExample
+          focusTrapOptions={{
+            onActivate,
+            onPostActivate,
+          }}
+        />
+      );
+
+      expect(onActivate).not.toHaveBeenCalled();
+      expect(onPostActivate).not.toHaveBeenCalled();
+
+      // Activate the focus trap
+      const activateTrapButton = screen.getByText('activate trap');
+      activateTrapButton.focus();
+      fireEvent.click(activateTrapButton);
+
+      expect(onActivate).toHaveBeenCalled();
+      expect(onPostActivate).toHaveBeenCalled();
+    });
+
+    it('Will call onPostDeactivate() even if checkCanReturnFocus() is undefined', async () => {
+      const onDeactivate = jest.fn();
+      const onPostDeactivate = jest.fn();
+
+      render(
+        <FocusTrapExample
+          focusTrapOptions={{
+            onDeactivate,
+            onPostDeactivate,
+          }}
+        />
+      );
+
+      // Activate the focus trap
+      const activateTrapButton = screen.getByText('activate trap');
+      activateTrapButton.focus();
+      fireEvent.click(activateTrapButton);
+
+      // Auto-sets focus inside the focus trap
+      await waitFor(() => {
+        expect(screen.getByText('Link 1')).toHaveFocus();
+      });
+
+      expect(onDeactivate).not.toHaveBeenCalled();
+      expect(onPostDeactivate).not.toHaveBeenCalled();
+
+      // Deactivate the focus trap
+      fireEvent.click(screen.getByText('deactivate trap'));
+
+      expect(onDeactivate).toHaveBeenCalled();
+      expect(onPostDeactivate).toHaveBeenCalled();
+    });
+
+    it('Will call onPostDeactivate() even if returnFocusOnDeactivate is false', async () => {
+      const onDeactivate = jest.fn();
+      const onPostDeactivate = jest.fn();
+
+      render(
+        <FocusTrapExample
+          focusTrapOptions={{
+            onDeactivate,
+            onPostDeactivate,
+            returnFocusOnDeactivate: false,
+          }}
+        />
+      );
+
+      // Activate the focus trap
+      const activateTrapButton = screen.getByText('activate trap');
+      activateTrapButton.focus();
+      fireEvent.click(activateTrapButton);
+
+      // Auto-sets focus inside the focus trap
+      await waitFor(() => {
+        expect(screen.getByText('Link 1')).toHaveFocus();
+      });
+
+      expect(onDeactivate).not.toHaveBeenCalled();
+      expect(onPostDeactivate).not.toHaveBeenCalled();
+
+      // Deactivate the focus trap
+      fireEvent.click(screen.getByText('deactivate trap'));
+
+      expect(onDeactivate).toHaveBeenCalled();
+      expect(onPostDeactivate).toHaveBeenCalled();
+    });
+
+    ['string', 'element', 'function'].forEach((elementSelectionMethod) => {
+      it(`Will return focus to setReturnFocus target, setReturnFocus type = ${elementSelectionMethod}`, async () => {
```

Comment:
Thanks for adding this!

Reply:
Kind of didn't have much choice with the code coverage restriction on PRs for this repo